x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_setup_state.tsx

```diff
@@ -4,52 +4,88 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { useState, useCallback, useMemo } from 'react';
+import { useState, useCallback, useMemo, useEffect } from 'react';
 
 import { isExampleDataIndex } from '../../../../common/log_analysis';
+import { ValidationIndicesError } from '../../../../common/http_api';
+import { useTrackedPromise } from '../../../utils/use_tracked_promise';
+import { callIndexPatternsValidate } from './api/index_patterns_validate';
 
 type SetupHandler = (
   indices: string[],
   startTime: number | undefined,
   endTime: number | undefined
 ) => void;
 
+export type ValidationIndicesUIError =
+  | ValidationIndicesError
+  | { error: 'NETWORK_ERROR' }
+  | { error: 'TOO_FEW_SELECTED_INDICES' };
+
+export interface ValidatedIndex {
+  index: string;
+  validation?: ValidationIndicesError;
```

Comment:
empty as in "no documents"? I'm not validating for that. Does it prevent the ML job/datafeed from being started?

That would be the only error that could happen alongside any other error.

Reply:
True, "empty" doesn't cause a problem. But it could lack both the timestamp and  a dataset fields, for example. I just mean that the API could return multiple errors for an index which we can't represent here.

Reply:
That's right. If we validate more than one field we can return more than one error.

Changed it already in the last push